patches/chromium/disable_use_lld_for_macos.patch

```diff
@@ -0,0 +1,21 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: mlaurencin <mlaurencin@electronjs.org>
+Date: Fri, 6 Aug 2021 15:29:48 -0700
+Subject: disable use_lld for macOS
+
+This patch disables use_lld on macOS, in order to prevent a linking bug
+that occurs when building for arm64.
+
+diff --git a/build/config/compiler/compiler.gni b/build/config/compiler/compiler.gni
+index 4a714ab0a5cdafded76bba553205f94599c27a30..d2723dd224a8afd2b6dd99ec11e9e71b7f76fe4a 100644
+--- a/build/config/compiler/compiler.gni
++++ b/build/config/compiler/compiler.gni
+@@ -201,7 +201,7 @@ declare_args() {
+   # In late bring-up on macOS (see docs/mac_lld.md), and not functional at all for
+   # iOS. The default linker everywhere else.
+   use_lld =
+-      is_clang && (!is_apple || (target_os == "mac" && chrome_pgo_phase != 1))
```

Comment:
This is a public flag, you can add it to our `build/args/all.gn` rather than a patch.

Reply:
@deepak1556 We want to disable it _only_ on macOS.  Which isn't supported afaik from `all.gn` / gn argument files (they don't have `is_mac` or `host_os`)

Reply:
Can't we do it from here https://github.com/electron/electron/blob/66c458a3530ed857955eb285700156b00bfc0271/.circleci/config.yml#L125

Reply:
yes but that won't fix it for folks building locally

Reply:
Oh the error happens locally as well ? I thought it was only a ci issue. In that case, do we have it reported to upstream ?

Reply:
I am okay with the patch, just a comment with a upstream bug link would go before this PR got merged.

Reply:
Can confirm the issue also happens in local build. @mlaurencin doesn't have an m1 device to follow-up with upstream. I don't want to block this PR, will create an issue upstream.